numpy/ma/core.py

```diff
@@ -3030,6 +3030,7 @@ def __array_wrap__(self, obj, context=None):
         if context is not None:
             result._mask = result._mask.copy()
             (func, args, _) = context
+            args = args[:func.nin]
```

Comment:
So, this is my worry: if even in `numpy`, we need this fix for `__array_wrap__`, we are likely going to break other classes as well. Do any tests break if we remove the output arguments in `__array_prepare__`?

Reply:
Remember that this part of the patch stands alone as a bug fix for np.add(a, b, c) currently inheriting the mask of c.

Any classes that we break are necessarily already broken in some calling convention in 1.14.

Omitting output arguments is possibly safest simply since passing them in the first place is uncommon.

Reply:
Ah, yes, I forgot that `__array_wrap__` does get them if put as a positional argument. I wonder if it makes more sense to do this fix in a separate PR, so we can disentangle it from the larger reworking (which could them remove this again if we go for no outputs at all).

Reply:
Yep, this seems like an uncontroversial fix - i'll make a separate patch for these lines